docs/core-concepts/index.md

```diff
@@ -0,0 +1,32 @@
+# Core Concepts
+
+These are some of the core concepts that you should be familiar with before going over the [Getting Started Guide](get-started/index.md).
+
+
+## GitOps
+GitOps is a way of managing your infrastructure and applications so that whole system is describe declaratively and version controlled(most likely in a git repository), and having an automated process that ensure that the deployed environment matches the state specified in a repository.
+
+For more information, take a look at [this documentation](https://gitops-community.github.io/kit/#what-is-gitops).
+
+## Git Source
+The git source represents a git repository in Flux.
+
+For more information, take a look at [this documentation](components/source/gitrepositories.md)
+
+## Kustomization
+The kustomization represents an application that Flux is supposed to reconcile in the cluster. It specifies a path in the git repository containing the `kustomization.yaml` for the application. If the repository contains only plain kubernetes manifests, a `kustomization.yaml` is generated for you. If there are non-kubernetes related files or manifest that you don't want to apply to the cluster, you can exclude it under the `spec.ignore` fields.
+
+For more information, take a look at [this documentation](components/kustomize/kustomization.md)
+
+## Reconciliation
+Reconcilation refers to ensuring the state(different resources e.g deployments) of a particular application(represented by a kustomization) in the cluster matches the state defined in a git repository. It performs this reconciliation every one minute by default but this can be specified in the kustomization. If you make any changes to the cluster using `kubectl edit` or `kubectl patch`, it will be promptly reverted. You either suspend the reconciliation or push your changes to a git repository.
+
+For more information, take a look at [this documentation](components/kustomize/kustomization.md#reconciliation)
+
+## Bootstrap
+The process of installing the Flux components in a complete GitOps way is called a bootstrap. The manifests are applied to the cluster, a git source and kustomization is created for the Flux component and the manifest are pushed to an existing git repository(or a new one is created). Flux can manage itself jst as it manages other applications. 
+The bootstrap is done using the flux CLI `flux bootstrap`.
```

Comment:
```suggestion
The bootstrap is done using the Flux CLI `flux bootstrap`.
```

or

```suggestion
The bootstrap is done using the `flux` CLI `flux bootstrap`.
```

I would maybe also include a link to the `flux bootstrap` documentation page.